core/src/main/scala/cats/arrow/FunctionK.scala

```diff
@@ -13,7 +13,7 @@ import cats.data.{EitherK, Tuple2K}
  *   val listToOption = λ[FunctionK[List, Option]](_.headOption)
  * }}}
  */
-trait FunctionK[F[_], G[_]] extends Serializable { self =>
+trait FunctionK[F[_], G[_]] extends PolyFunction with Serializable { self =>
```

Comment:
While extending PolyFunction should work I wouldn't really recommend it given that it's likely under-tested and we're still figuring out how these things should work exactly, so if there's a problem you'll be stuck with it due to the binary-compatibility requirements of cats. This also won't be of much use once we have polymorphic SAMs. Also it might make more sense to redefine `type FunctionK[F[_], G[_]] = [A] => F[A] => G[A]` whenever you'll be able to break binary-compatibility.